src/main/scala/tile/FPU.scala

```diff
@@ -241,9 +241,10 @@ trait HasFPUParameters {
   val (dExpWidth, dSigWidth) = (FType.D.exp, FType.D.sig)
   val floatTypes = FType.all.filter(_.ieeeWidth <= fLen)
   val maxType = floatTypes.last
-  def prevType(t: FType) = floatTypes(floatTypes.indexOf(t) - 1)
+  def prevType(t: FType) = floatTypes(typeTag(t) - 1)
   val maxExpWidth = maxType.exp
   val maxSigWidth = maxType.sig
+  def typeTag(t: FType) = floatTypes.indexOf(t)
```

Comment:
`typeTag` makes me think of http://docs.scala-lang.org/overviews/reflection/typetags-manifests.html, but it looks like we are no longer using the scala construct anywhere in the code base, so...ymmv.